backend/tests/v5/e2e/routes/sso/aad/aad.test.js

```diff
@@ -0,0 +1,180 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const SuperTest = require('supertest');
+const ServiceHelper = require('../../../../helper/services');
+const { src } = require('../../../../helper/path');
+const { generateRandomString } = require('../../../../helper/services');
+const { authenticateRedirectUri, signupRedirectUri } = require('../../../../../../src/v5/services/sso/aad/aad.constants');
+
+jest.mock('../../../../../../src/v5/services/sso/aad', () => ({
+	...jest.requireActual('../../../../../../src/v5/services/sso/aad'),
+	getUserDetails: jest.fn(),
+}));
+const Aad = require('../../../../../../src/v5/services/sso/aad');
+const { aad } = require('../../../../../../src/v5/services/sso/sso.constants');
+
+const { templates } = require(`${src}/utils/responseCodes`);
+
+let server;
+let agent;
+
+const userEmail = 'example@email.com';
+const testUser = ServiceHelper.generateUserCredentials();
+const userEmailSso = 'example2@email.com';
+const testUserSso = ServiceHelper.generateUserCredentials();
+const newUserEmail = 'newUserEmail@email.com';
+
+const setupData = async () => {
+	await ServiceHelper.db.createUser(testUser, [], { email: userEmail });
+	await ServiceHelper.db.createUser(testUserSso, [], { email: userEmailSso,
+		sso: { type: aad, id: generateRandomString() } });
+};
+
+const testAuthenticate = () => {
+	describe('Sign Up Authenticate', () => {
+		test('should redirect the user to Microsoft authentication page', async () => {
+			const signupUri = generateRandomString();
+			const res = await agent.get(`/v5/sso/aad/authenticate?signupUri=${signupUri}`)
+				.expect(templates.found.status);
+
+			const redirectUri = res.headers.location;
+			expect(redirectUri).toEqual(
```

Comment:
parse the URL and test for the parts you want to test https://nodejs.org/docs/latest-v14.x/api/url.html#url_the_whatwg_url_api